byway/src/main/java/com/google/sps/data/Login.java

```diff
@@ -0,0 +1,25 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+public final class Login{ 
```

Comment:
When you have a plain data class with all final fields, where you're not going to change fields in the type after its constructed, it's good to annotate it with [`@Immutable`](https://errorprone.info/bugpattern/Immutable). This way the ErrorProne checker will help you make sure you don't actually make something mutable without intending to. 

Immutable objects are nice in complex systems because they're always safe to share across threads and re-use without needing to make copies or worry about other threads or other parts of the program modifying the data unexpectedly.

Reply:
done